test/node-rpc-test.js

```diff
@@ -225,4 +239,148 @@ describe('RPC', function() {
+    const KEEP_BLOCKS = network.KEEP_BLOCKS;
+
+    let nclient, node;
+
+    before(() => {
+      network.block.pruneAfterHeight = 10;
+      network.block.keepBlocks = 10;
+    });
+
+    after(() => {
+      network.block.pruneAfterHeight = PRUNE_AFTER_HEIGHT;
+      network.block.keepBlocks = KEEP_BLOCKS;
+    });
+
+    afterEach(async () => {
+      if (nclient && nclient.opened)
+        await nclient.close();
+
+      if (node && node.opened)
+        await node.close();
+    });
+
+    it('should fail with wrong arguments', async () => {
+      node = new FullNode(nodeOptions);
+      nclient = new NodeClient(clientOptions);
+
+      await node.open();
+
+      await assert.rejects(async () => {
+        await nclient.execute('pruneblockchain', [1]);
+      }, {
+        code: errs.MISC_ERROR,
+        type: 'RPCError',
+        message: 'pruneblockchain'
+      });
+
+      await node.close();
+    });
+
+    it('should not work for spvnode', async () => {
+      node = new SPVNode(nodeOptions);
+      nclient = new NodeClient(clientOptions);
+
+      await node.open();
+
+      await assert.rejects(async () => {
+        await nclient.execute('pruneblockchain');
+      }, {
+        type: 'RPCError',
+        message: 'Cannot prune chain in SPV mode.',
+        code: errs.MISC_ERROR
+      });
+
+      await node.close();
+    });
+
+    it('should fail for pruned node', async () => {
+      node = new FullNode({
+        ...nodeOptions,
+        prune: true
+      });
+
+      await node.open();
+
+      await assert.rejects(async () => {
+        await nclient.execute('pruneblockchain');
+      }, {
+        type: 'RPCError',
+        code: errs.MISC_ERROR,
+        message: 'Chain is already pruned.'
+      });
+
+      await node.close();
+    });
+
+    it('should fail for short chain', async () => {
+      node = new FullNode(nodeOptions);
+
+      await node.open();
+
+      await assert.rejects(async () => {
+        await nclient.execute('pruneblockchain');
+      }, {
+        type: 'RPCError',
+        code: errs.MISC_ERROR,
+        message: 'Chain is too short for pruning.'
+      });
+
+      await node.close();
+    });
+
+    it('should prune chain', async () => {
+      // default - prune: false
+      node = new FullNode(nodeOptions);
+      nclient = new NodeClient(clientOptions);
+
+      await node.open();
+
+      const addr = 'rs1q4rvs9pp9496qawp2zyqpz3s90fjfk362q92vq8';
+      node.miner.addAddress(addr);
+
+      // generate 30 blocks.
+      // similar to chain-rpc-test
+      const blocks = await nclient.execute('generate', [30]);
+
+      // make sure we have all the blocks.
+      for (let i = 0; i < 30; i++) {
+        const block = await nclient.execute('getblock', [blocks[i]]);
+        assert(block);
+      }
+
+      // now prune..
+      await nclient.execute('pruneblockchain');
+
+      // behined height check
+      for (let i = 0; i < 10; i++) {
```

Comment:
for these tests instead of using `10` you might want to use the constants so future reviewers understand what the number represents (`PRUNE_AFTER_HEIGHT` and `KEEP_BLOCKS`)

Reply:
☝️ this goes for the chain test too